lib/lambda-worker/lambda-worker.ts

```diff
@@ -0,0 +1,199 @@
+import * as cdk from "@aws-cdk/core";
+import * as cloudwatch from "@aws-cdk/aws-cloudwatch";
+import * as cloudwatchActions from "@aws-cdk/aws-cloudwatch-actions";
+import * as eventSource from "@aws-cdk/aws-lambda-event-sources";
+import * as lambda from "@aws-cdk/aws-lambda";
+import * as lambdaNodeJs from "@aws-cdk/aws-lambda-nodejs";
+import * as sqs from "@aws-cdk/aws-sqs";
+import * as subs from "@aws-cdk/aws-sns-subscriptions";
+
+import { LambdaWorkerProps } from "./lambda-worker-props";
+
+const DEFAULT_MAX_RECEIVE_COUNT = 5;
+const DEFAULT_APPROX_AGE_OLDEST_MESSAGE_THRESHOLD = cdk.Duration.hours(1);
+const DEFAULT_APPROX_NUM_MESSAGES_VISIBLE_THRESHOLD = 1000;
+const MINIMUM_MEMORY_SIZE = 1024;
+const MINIMUM_LAMBDA_TIMEOUT = cdk.Duration.seconds(30);
+
+export class LambdaWorker extends cdk.Construct {
+  constructor(scope: cdk.Construct, id: string, props: LambdaWorkerProps) {
+    super(scope, id);
+
+    // Lambda settings
+    const memorySize =
+      props.lambdaProps.memorySize &&
+      props.lambdaProps.memorySize > MINIMUM_MEMORY_SIZE
```

Comment:
I'd be inclined to throw an error here, rather than self-healing. Better to know before you commit your code than be surprised after it's up and running.